cli/command/cli.go

```diff
@@ -280,7 +286,7 @@ func newAPIClientFromEndpoint(ep docker.Endpoint, configFile *configfile.ConfigF
 }
 
 func resolveDockerEndpoint(s store.Store, contextName string, opts *cliflags.CommonOptions) (docker.Endpoint, error) {
-	if contextName != "" {
+	if contextName != "" || contextName == DefaultContextName {
```

Comment:
Is this logic flipped? i.e. should this be

```suggestion
	if contextName != "" && contextName != DefaultContextName {
```


But perhaps better to return early for the special case;

```suggestion
	if contextName == "" || contextName == DefaultContextName {
		return resolveDefaultDockerEndpoint(opts)
	}
```